pkg/sql/logictest/testdata/logic_test/create_table

```diff
@@ -110,3 +110,208 @@ WHERE feature_name IN (
 ----
 sql.schema.inverted_index
 sql.schema.hash_sharded_index
+
+subtest like_table
+
+# Test the CREATE TABLE LIKE functionality.
+
+statement ok
+CREATE TABLE like_table (
+  a INT CHECK (a > 3),
+  b STRING DEFAULT 'foo' NOT NULL,
+  c DECIMAL AS (a+3) STORED,
+  h INT,
+  j JSON,
+  PRIMARY KEY (a, b),
+  UNIQUE INDEX foo (b DESC, c),
+  INDEX (c) STORING (j),
+  INVERTED INDEX (j)
+)
+
+statement ok
+CREATE TABLE like_none (LIKE like_table)
+
+query TT
+SHOW CREATE TABLE like_none
+----
+like_none  CREATE TABLE like_none (
+           a INT8 NOT NULL,
+           b STRING NOT NULL,
+           c DECIMAL NULL,
+           h INT8 NULL,
+           j JSONB NULL,
+           FAMILY "primary" (a, b, c, h, j, rowid)
+)
+
+statement ok
+CREATE TABLE like_constraints (LIKE like_table INCLUDING CONSTRAINTS)
+
+query TT
+SHOW CREATE TABLE like_constraints
+----
+like_constraints  CREATE TABLE like_constraints (
+                  a INT8 NOT NULL,
+                  b STRING NOT NULL,
+                  c DECIMAL NULL,
+                  h INT8 NULL,
+                  j JSONB NULL,
+                  FAMILY "primary" (a, b, c, h, j, rowid),
+                  CONSTRAINT check_a CHECK (a > 3)
+)
+
+statement ok
+CREATE TABLE like_indexes (LIKE like_table INCLUDING INDEXES)
```

Comment:
is there any wierd behavior with having a unique constraint + an index and then saying like including indexes and constraints? Basically looking at the case where the constraints and indexes overlap